artiq/test/coredevice/test_embedding.py

```diff
@@ -117,6 +117,89 @@ def test_default_arg(self):
         self.assertEqual(exp.run(), 42)
 
 
+class _RPCOverflowTest(EnvExperiment):
+    def build(self):
+        self.setattr_device("core")
+
+    @kernel
+    def run(self, f, g):
+        f(g())
+
+
+class RPCOverflowTest(ExperimentCase):
+    def assert_roundtrip_value_equal(self, g):
+        exp = self.create(_RPCOverflowTest)
+        exp.run(lambda x: self.assertEqual(g(), x), g)
+
+    def test_i32(self):
+        def g() -> TInt32:
+            return (2 ** 31) - 1
+
+        x = randint(-2 ** 31, (2 ** 31) - 1)
+
+        def h() -> TInt32:
+            return x
+
+        self.assert_roundtrip_value_equal(g)
+        self.assert_roundtrip_value_equal(h)
+
+    def test_i64(self):
+        def g() -> TInt64:
+            return (2 ** 63) - 1
+
+        x = randint(-2 ** 63, (2 ** 63) - 1)
+
+        def h() -> TInt64:
+            return x
+
+        self.assert_roundtrip_value_equal(g)
+        self.assert_roundtrip_value_equal(h)
+
+    def test_array_i32(self):
+        x = randint(-2 ** 31, (2 ** 31) - 1)
```

Comment:
Where is randint defined? Did you test this?
Also, deterministic tests are better.

Reply:
thanks for the reminder!